src/emmy/expression/compile.cljc

```diff
@@ -369,6 +373,25 @@
               (gensym-fn 'y)))]
     (rec state)))
 
+(defn- cse
+  "Invokes extract-common-subexpressions on `x`.
+
+   The results of the elimination are presented in a map with the keys
+   `:bindings` and `body`.  If no suitable subexpressions were found,
+   the binding list will be empty and body will be unchanged.
+
+   "
+  [x opts]
+  (extract-common-subexpressions
+   x
+   (fn [body bindings]
+     ;; TODO: consistency demands that we do apply-numeric-ops on the
+     ;; intermediate bindings, but that will shake up a lot of unit
+     ;; tests so I'm temporarily postponing this.
+     {:bindings bindings #_(into {} (for [[k v] bindings] [k (apply-numeric-ops v)]))
```

Comment:
oh, let's be careful to actually NOT put these into a map, since that might mix up the ordering, which definitely matters here. I think these bindings need to stay a sequence of pairs. Is that how they come into the callback?

Reply:
😮 you're right. Thank you. I forgot about this

Reply:
done